app/model_config.py

```diff
@@ -0,0 +1,173 @@
+import os
+
+# This is directory tree structure for models and their metadata inside of
+# Google Cloud Storage. The top-level item (e.g. "bert_classification_models")
+# is called the 'bucket'.
+# .
+# └── bert_classification_models
+#     └── multilabel
+#         └── UPR_2percent_ps0
+#             └── saved_model
+#                 ├── label.vocab
+#                 └── 1573031002
+#                     ├── saved_model.pb
+#                     └── variables
+#                         ├── variables.data-00000-of-00001
+#                         └── variables.index
+#         └── UPR_2percent_ps512
+#             └── saved_model
+#                 ├── label.vocab
+#                 └── 1572866595
+#                     ├── saved_model.pb
+#                     └── variables
+#                         ├── variables.data-00000-of-00001
+#                         └── variables.index
+
+
+IN = {
+    "bucket_name": "bert_classification_models",
+    "model_name": "UPR_2percent_ps0",
+    "instance_name": "1573031002",
+    "saved_model": {
+        "name": "saved_model.pb",
+        "path": "multilabel/{model}/saved_model/{instance}",
+    },
+    "variables": {
+        "name": "*",
+        # TODO: come up with a convention on handling recursive copying in the
+        # input configuration.
+        "path": "multilabel/{model}/saved_model/{instance}/variables",
+    },
+    "vocab": {
+        "name": "label.vocab",
+        "path": "multilabel/{model}/saved_model",
+    },
+}
+
+
+class ModelConfig(object):
```

Comment:
not sure how this class is better than a pure dictionary...

Reply:
It allows for much more concise and auto-completable :) access to attributes using @property, which additionally allows for constructed properties like dir, file and fully-qualified file name without needing to hardcode the concatenation in each path dict.

os.path.join(IN["saved_model"]["path"], IN["saved_model"]["name"]).format(model=model, instance=instance) 

vs. 

src.saved_model.fqfn

I don't love this either though. Let me keep thinking about this one.

Reply:
What you say is true of PathConfig, but I don't see what exactly ModelConfig does other than provide a method to access a dictionary...